tensorflow/python/kernel_tests/array_ops_test.py

```diff
@@ -178,8 +177,12 @@ def testCompare(self):
       self._compare(4, t, False)
       self._compare(5, t, False)
 
-    with self.assertRaises(ValueError):
-      self._compare(6, np.float32, False)
+    # Test for inputs with rank not equal to 1
+    x = [[1, 1], [1, 1]]
+    with self.assertRaises(errors.InvalidArgumentError):
```

Comment:
Use `self.assertRaisesRegexp` to verify you're catching the expected error string.